app/js/validateQuestionsFormData.js

```diff
@@ -0,0 +1,180 @@
+
+    /**
+     * IF both answer1 and answer2 fields are filled in, populate newQuestion obj and hide both errors :D
+     */
+    function validAnswer1and2() {
+        if(!oneIsWrong && !twoIsWrong) {
+            hideErrorsForAnswer(answer1Error)
+            hideErrorsForAnswer(answer2Error)
+            populateQuestionMinimumAnswers()
+        }
+    }
+
+    /**
+     * IF answer1 field is blank but answer2 is filled in, will show error for answer1
+     */
+    function invalidAnswer1() {
+        if(oneIsWrong && !twoIsWrong) {
+            hideErrorsForAnswer(answer2Error)
+            showErrorsForAnswer(answer1Error)
+        }
+    }
+
+    /**
+     * IF answer2 field is blank but answer1 is filled in, will show error for answer2
+     */
+    function invalidAnswer2() {
+        if(!oneIsWrong && twoIsWrong) {
+            hideErrorsForAnswer(answer1Error)
+            showErrorsForAnswer(answer2Error)
+        }
+    }
+
+    /**
+     * IF answer1 and answer2 fields are blank will show both errors
+     */
+    function invalidAnswers() {
+        if(oneIsWrong && twoIsWrong) {
+            showErrorsForAnswer(answer1Error)
+            showErrorsForAnswer(answer2Error)
+        }
+    }
+
+    /**
+     * Trims whitespace and checks answer exists,
+     * Populates newQuestion object with answer value and assigned key
+     *
+     * @param answer - answer value (eg: answer4 query selector variable)
+     * @param key - option value (eg: 'option4')
+     */
+    function validateOptionalField(answer, key) {
+        if (answer.trim().length !==0) {
+            newQuestion[key] = answer
+        }
+    }
+
+    /**
+     * Validates optional answers for answer field 3, 4 and 5
+     */
+    function validateOptionalAnswers() {
+        validateOptionalField(answer3, 'option3')
+        validateOptionalField(answer4, 'option4')
+        validateOptionalField(answer5, 'option5')
+    }
+
+    /**
+     * ForEach validation loop over radio buttons;
+     * if one is checked - populates newQuestion object with answer reference value
+     *
+     * @returns {boolean} - if radio button checked = true
+     * @returns {boolean} - if radio button is unchecked = false
+     */
+    function setCorrectAnswer() {
+        correctAnswers.forEach( function(answer) {
+            if (answer.checked) {
+                newQuestion.answer = answer.value
+            }
+        })
+        return newQuestion.answer;
+    }
+
+    /**
+     * Removes error message if answer is correctly validated
+     */
+    function validateCorrectAnswer() {
+        if (!setCorrectAnswer()) {
+            correctAnswerError.classList.remove('hidden')
+        } else {
+            correctAnswerError.classList.add('hidden')
+        }
+    }
+  
+    validateQuestionText()
+    validAnswer1and2()
+    invalidAnswer1()
+    invalidAnswer2()
+    invalidAnswers()
+    validateOptionalAnswers()
+    validateCorrectAnswer()
+    postQuestionsEdit(newQuestion)
```

Comment:
this sends the data to the API even if the validation fails, it shows the message but it also shows the failure message, you want to only send the data if the validation has passed